src/Language/Syntax.hs

```diff
@@ -85,53 +62,66 @@ data BoardEq a = PosDef
 instance Functor BoardEq where
   fmap f (PosDef n p1 p2 e) = PosDef n p1 p2 (fmap f e)
 
-instance Show (BoardEq a) where
-   show (PosDef n x y e) = n ++ "(" ++ show x ++ ", " ++ show y ++ ")" ++ " = " ++ show e
-
 data Pos = Index Int
          | ForAll Name
          deriving (Eq, Generic)
 
-instance Show Pos where
-   show (Index i) = show i
-   show (ForAll n) = n
-
 instance Ord Pos where
-  compare (Index i) (Index j)      = compare i j
-  compare (ForAll _) (Index i)     = if i <= 0 then GT else LT
+  compare (Index i) (Index j)       = compare i j
+  compare (ForAll _) (Index i)      = if i <= 0 then GT else LT
   compare ix@(Index i) f@(ForAll _) = compare f ix
-  compare (ForAll _) (ForAll _)    = EQ
-
- -- | Expressions
-data Expr a = I Int                     -- ^ Integer expression
-          | S Name                        -- ^ Symbol
-          | B Bool                        -- ^ Boolean
-          | Ref Name                      -- ^ Reference to a variable
-          | Tuple [Expr a]                  -- ^ Tuple of 'Expr'
-          | App Name (Expr a)                -- ^ Application of the function called Name to the tuple of arguments
-          | Binop Op (Expr a) (Expr a)            -- ^ Binary operation of two expressions
-          | Let Name (Expr a) (Expr a)           -- ^ Let binding
-          | If (Expr a) (Expr a) (Expr a)             -- ^ Conditional expression
-          | While (Expr a) (Expr a) [Name] (Expr a)   -- ^ While: condition, body, names of arguments from the wrapper function, (tuple of) expression(s) which referenc(es) the name(s).
-          -- the last Expr can always be constructed from the [Name], but it makes the code cleaner to do that only once while parsing
-          | Annotation a (Expr a) -- ^ Parameterized by the type of an annotation (could use a cofree comonad, but also this)
-          | HE Name -- ^ Type hole
+  compare (ForAll _) (ForAll _)     = EQ
+
+-- | Expressions
+data Expr a = I Int                                 -- ^ Integer
+          | S Name                                  -- ^ Symbol
+          | B Bool                                  -- ^ Boolean
+          | Ref Name                                -- ^ Reference to a variable
+          | Tuple [Expr a]                          -- ^ Tuple of 'Expr'
+          | App Name (Expr a)                       -- ^ Function application with argument tuple
+          | Binop Op (Expr a) (Expr a)              -- ^ Binary operation
+          | Let Name (Expr a) (Expr a)              -- ^ Let binding
+          | While (Expr a) (Expr a) [Name] (Expr a) -- ^ While loop
+          | If (Expr a) (Expr a) (Expr a)           -- ^ Conditional
+          | Annotation a (Expr a)                   -- ^ Parameterized by the type of an annotation
+          | HE Name                                 -- ^ Type hole
```

Comment:
Nice, yeah this looks a lot cleaner, I'm a big fan of keeping the comments indent aligned